index.js

```diff
@@ -0,0 +1,1285 @@
+
+  //  Arguments$prototype$equals :: Arguments ~> Arguments -> Boolean
+  var Arguments$prototype$equals = function(other) {
+    return Array$prototype$equals.call(this, other);
+  };
+
+  //  Error$prototype$toString :: Error ~> () -> String
+  var Error$prototype$toString = function() {
+    return 'new ' + this.name + '(' + toString(this.message) + ')';
+  };
+
+  //  Error$prototype$equals :: Error ~> Error -> Boolean
+  var Error$prototype$equals = function(other) {
+    return equals(other.name, this.name) &&
+           equals(other.message, this.message);
+  };
+
+  //  Object$empty :: () -> StrMap a
+  var Object$empty = function() {
+    return {};
+  };
+
+  //  Object$prototype$toString :: StrMap a ~> () -> String
+  var Object$prototype$toString = function() {
+    var reprs = [];
+    var keys = Object.keys(this).sort();
+    for (var idx = 0; idx < keys.length; idx += 1) {
+      var k = keys[idx];
+      reprs.push(toString(k) + ': ' + toString(this[k]));
+    }
+    return '{' + reprs.join(', ') + '}';
+  };
+
+  //  Object$prototype$equals :: StrMap a ~> StrMap a -> Boolean
+  var Object$prototype$equals = function(other) {
+    var self = this;
+    var keys = Object.keys(this).sort();
+    return equals(Object.keys(other).sort(), keys) &&
+           keys.every(function(k) { return equals(other[k], self[k]); });
+  };
+
+  //  Object$prototype$concat :: StrMap a ~> StrMap a -> StrMap a
+  var Object$prototype$concat = function(other) {
+    var result = {};
+    for (var k in this) result[k] = this[k];
+    for (k in other) result[k] = other[k];
+    return result;
+  };
+
+  //  Object$prototype$map :: StrMap a ~> (a -> b) -> StrMap b
+  var Object$prototype$map = function(f) {
+    var result = {};
+    for (var k in this) result[k] = f(this[k]);
+    return result;
+  };
+
+  //  Object$prototype$reduce :: StrMap a ~> ((b, a) -> b, b) -> b
+  var Object$prototype$reduce = function(f, initial) {
+    var result = initial;
+    for (var k in this) result = f(result, this[k]);
+    return result;
+  };
+
+  //  Function$of :: b -> (a -> b)
+  var Function$of = function(x) {
+    return function(_) { return x; };
+  };
+
+  //  Function$prototype$map :: (a -> b) ~> (b -> c) -> (a -> c)
+  var Function$prototype$map = function(f) {
+    var functor = this;
+    return function(x) { return f(functor(x)); };
+  };
+
+  //  Function$prototype$promap :: (b -> c) ~> (a -> b, c -> d) -> (a -> d)
+  var Function$prototype$promap = function(f, g) {
+    var profunctor = this;
+    return function(x) { return g(profunctor(f(x))); };
+  };
+
+  //  Function$prototype$ap :: (a -> b) ~> (a -> b -> c) -> (a -> c)
+  var Function$prototype$ap = function(f) {
+    var apply = this;
+    return function(x) { return f(x)(apply(x)); };
+  };
+
+  //  Function$prototype$chain :: (a -> b) ~> (b -> a -> c) -> (a -> c)
+  var Function$prototype$chain = function(f) {
+    var chain = this;
+    return function(x) { return f(chain(x))(x); };
+  };
```

Comment:
Fortuitously, @buzzdecafe opened ramda/ramda#1936 today which alerted me to the fact that Function can satisfy the requirements of Chain. The implementation is straightforward. In fact, it's almost identical to that of `Function$prototype$ap` since the only difference between the two functions is that one takes `(a -> b -> c)` whereas the other takes `(b -> a -> c)`.